### PR TITLE
[Extensions] Added tests for PriorityCache, ADTaskProfileTransportActionTests and increased code coverage

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
First PR to add unit tests back for AD Extension.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-sdk-java/issues/723

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/settings/EnabledSetting.java [hunk 1]
```diff
@@ -39,8 +39,9 @@ public class EnabledSetting extends AbstractSetting {
 
     public static final String LEGACY_OPENDISTRO_AD_BREAKER_ENABLED = "opendistro.anomaly_detection.breaker.enabled";
 
-    public static final String INTERPOLATION_IN_HCAD_COLD_START_ENABLED =
-        "plugins.anomaly_detection.hcad_cold_start_interpolation.enabled";;
+    public static final String INTERPOLATION_IN_HCAD_COLD_START_ENABLED = "plugins.anomaly_detection.hcad_cold_start_interpolation.enabled";
+
+    public static final String DOOR_KEEPER_IN_CACHE_ENABLED = "plugins.anomaly_detection.door_keeper_in_cache.enabled";;
```
Comment (saratvemulapalli) [inline src/main/java/org/opensearch/ad/settings/EnabledSetting.java:44]: Is this new (merge from main)? 
I dont see it being used in the PR.

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java [hunk 2]
```diff
@@ -36,24 +89,22 @@ public void setUp() throws Exception {
...
-            Collections
-                .unmodifiableSet(
-                    new HashSet<>(
-                        Arrays
-                            .asList(
-                                AnomalyDetectorSettings.DEDICATED_CACHE_SIZE,
-                                AnomalyDetectorSettings.MODEL_MAX_SIZE_PERCENTAGE,
-                                AnomalyDetectorSettings.MODEL_MAX_SIZE_PERCENTAGE,
-                                AnomalyDetectorSettings.CHECKPOINT_TTL,
-                                AnomalyDetectorSettings.CHECKPOINT_SAVING_FREQ
```
Comment (saratvemulapalli) [inline src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:51]: Curious why these settings are not tested in the new changes?

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java [hunk 3]
```diff
@@ -656,24 +696,4 @@ public void testGetNoPriorityUpdate() {
...
-
-        assertTrue(null != entityCache.get(entity2.getModelId(detectorId).get(), detector));
-
-        entityCache.removeEntityModel(detectorId, entity2.getModelId(detectorId).get());
-
-        assertTrue(null == entityCache.get(entity2.getModelId(detectorId).get(), detector));
-
-        verify(checkpoint, times(1)).deleteModelCheckpoint(eq(entity2.getModelId(detectorId).get()), any());
-        verify(checkpointWriteQueue, never()).write(any(), anyBoolean(), any());
-    }
```
Comment (saratvemulapalli) [inline src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:677]: Why are these changes removed, I dont understand what they do (yet) but shouldn't we test them?

**Review by saratvemulapalli — COMMENTED** [review]
Great to see unit tests working for AD Extension. 

Can we start to uncomment code coverage checks in `build.gradle` for these files?

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java (hunk 2 above)
Comment (owaiskazi19) [reply src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:51]: These settings were added to `main` after we created `feature/extensions` [here](https://github.com/opensearch-project/anomaly-detection/pull/620/files#diff-00c99f214d22b6de9c5e592fc22fed35b542a8afc7f6b9d23c93e4efd0c2a5c0R103). Probably this test was fetched from `main` because it doesn't incline with the code changes of `feature/extensions`.

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java (hunk 3 above)
Comment (owaiskazi19) [reply src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:677]: Same as above

File: src/main/java/org/opensearch/ad/settings/EnabledSetting.java (hunk 1 above)
Comment (owaiskazi19) [reply src/main/java/org/opensearch/ad/settings/EnabledSetting.java]: Removed. This was coming from `main`

**Review by dbwiddis — APPROVED** [review]
Approving with the caveat that I think we're trying to catch up as much as we can with our current branch, but at some point will need to re-fork and that will catch up with other additions since then.

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java (hunk 2 above)
Comment (owaiskazi19) [reply src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:51]: These settings on feature/extensions are of the type [Duration](https://github.com/opensearch-project/anomaly-detection/blob/feature/extensions/src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java#L183) whereas based on the above PR the type got changed to [Setting](https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java#L196). Changing the type would require to fetch more code changes from `main` which we will be doing anyway with the re-fork.
The reason of the conflict here is the code on feature/extenions was from the time when we forked but probably while commenting the code for these tests, we pulled the code from main for the tests.

File: src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java [hunk 4]
```diff
@@ -20,7 +72,7 @@ public class PriorityCacheTests extends AbstractCacheTest {
     CheckpointDao checkpoint;
     ModelManager modelManager;
 
-    ClusterService clusterService;
+    SDKClusterService clusterService;
```
Comment (vibrantvarun) [inline src/test/java/org/opensearch/ad/caching/PriorityCacheTests.java:75]: clusterService -> sdkClusterService

**Review by joshpalis — APPROVED** [review]
